frontend/ui/network/manager.lua

```diff
@@ -532,6 +556,12 @@ function NetworkMgr:isOnline()
         return true
     end
 
+    -- Fail early if we don't even have a default route.
+    -- On PocketBook devices, if the first call to socket.dns.toip(…) fails, it never succeeds again.
+    if not self:hasDefaultRoute() then
```

Comment:
Given that this rare condition only happens on pocketbooks you could guard it behind a `if Device:isPocketBook()`

Reply:
> Done: [27052d4](https://github.com/koreader/koreader/commit/27052d4a56b96a09c6277183a43b49ac3b6d32fb)
> 
> What I could also do is factor out the `socket.dns.toip("dns.msftncsi.com") ~= nil` check into (say) `NetworkMgr:canResolveHostnames`, keep `isOnline` as it was, and only override it in `frontend/device/pocketbook/device.lua`:
> 
> ```lua
> function PocketBook:initNetworkManager(NetworkMgr)
>     …
>     function NetworkMgr:isOnline()
>         return self:hasDefaultRoute() and self:canResolveHostnames()
>     end
>     …
> end
> ```
> 
> But then, the whole point of `hasDefaultRoute` is that it's okay to be used on other platforms as well—it doesn't have false negatives as `isConnected` might have.

Honestly I feel this would be the nicest way to go. I would just duplicate the DNS check in the pocketbook code without changing the generic implementation. 

@NiLuJe what do you think?